src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -220,8 +238,8 @@ public AND(Condition a, Condition b) {
         }
 
         @Override
-        public boolean matches(IRoutingTile tile, EntityMinecart cart) {
-            return a.matches(tile, cart) && b.matches(tile, cart);
+        public int evaluate(IRoutingTile tile, EntityMinecart cart) {
+            return (a.evaluate(tile, cart) != 0 ? b.evaluate(tile, cart) : valueForBool(false));
```

Comment:
Sounds complicated. Keep it simple I say, and no breaking old tables.

However, this always returns the value of b. Is that desired?